src/main/java/io/confluent/connect/jdbc/dialect/CrateDatabaseDialect.java

```diff
@@ -0,0 +1,211 @@
+import io.confluent.connect.jdbc.sink.metadata.SinkRecordField;
+import io.confluent.connect.jdbc.util.ColumnId;
+import io.confluent.connect.jdbc.util.ExpressionBuilder;
+import io.confluent.connect.jdbc.util.ExpressionBuilder.Transform;
+import io.confluent.connect.jdbc.util.IdentifierRules;
+import io.confluent.connect.jdbc.util.TableId;
+
+/**
+ * A {@link DatabaseDialect} for Crate.
+ */
+public class CrateDatabaseDialect extends GenericDatabaseDialect {
+
+  /**
+   * The provider for {@link CrateDatabaseDialect}.
+   */
+  public static class Provider extends SubprotocolBasedProvider {
+    public Provider() {
+      super(CrateDatabaseDialect.class.getSimpleName(), "crate");
+    }
+
+    @Override
+    public DatabaseDialect create(AbstractConfig config) {
+      return new CrateDatabaseDialect(config);
+    }
+  }
+
+  /**
+   * Create a new dialect instance with the given connector configuration.
+   *
+   * @param config the connector configuration; may not be null
+   */
+  public CrateDatabaseDialect(AbstractConfig config) {
+    super(config, new IdentifierRules(".", "\"", "\""));
+  }
+
+  /**
+   * Perform any operations on a {@link PreparedStatement} before it is used. This is called from
+   * the {@link #createPreparedStatement(Connection, String)} method after the statement is
+   * created but before it is returned/used.
+   *
+   * <p>This method sets the {@link PreparedStatement#setFetchDirection(int) fetch direction}
+   * to {@link ResultSet#FETCH_FORWARD forward} as an optimization for the driver to allow it to
+   * scroll more efficiently through the result set and prevent out of memory errors.
+   *
+   * @param stmt the prepared statement; never null
+   * @throws SQLException the error that might result from initialization
+   */
+  @Override
+  protected void initializePreparedStatement(PreparedStatement stmt) throws SQLException {
+    log.trace("Initializing PreparedStatement fetch direction to FETCH_FORWARD for '{}'", stmt);
+    stmt.setFetchDirection(ResultSet.FETCH_FORWARD);
+  }
+
+  @Override
+  protected String getSqlType(SinkRecordField field) {
+    if (field.schemaName() != null) {
+      switch (field.schemaName()) {
+        case Decimal.LOGICAL_NAME:
+          return "DOUBLE";
+        case Date.LOGICAL_NAME:
+          return "TIMESTAMP";
+        case Time.LOGICAL_NAME:
+          return "TIMESTAMP";
+        case Timestamp.LOGICAL_NAME:
+          return "TIMESTAMP";
+        default:
+          // fall through to normal types
+      }
+    }
+    switch (field.schemaType()) {
+      case INT8:
+        return "INTEGER";
+      case INT16:
+        return "INTEGER";
+      case INT32:
+        return "INTEGER";
+      case INT64:
+        return "INTEGER";
+      case FLOAT32:
+        return "FLOAT";
+      case FLOAT64:
+        return "FLOAT";
+      case BOOLEAN:
+        return "BOOLEAN";
+      case STRING:
+        return "VARCHAR";
+      case BYTES:
+        return "BYTE";
```

Comment:
I'd be concerned about this mapping. CrateDB's `BYTE` type corresponds to a single byte, but the JDBC `BYTES` type really corresponds to an array of bytes.

Reply:
Could you give me CrateDB's data type references, it not specified in docs https://crate.io/docs/crate/reference/en/latest/general/ddl/data-types.html

Reply:
From that link:

> All types have the same ranges as corresponding [Java types](https://docs.oracle.com/javase/tutorial/java/nutsandbolts/datatypes.html).